hooks/mari/publish_textures.py

```diff
@@ -115,8 +117,86 @@ def validate(self, task_settings, item):
                 self.logger.error(error_msg)
                 return False
 
-        return super(MariPublishTexturesPlugin, self).validate(task_settings, item)
+        all_udims = {patch.udim() for patch in geo.patchList()}
+        selected_udims = {1001 + uv_index for uv_index in item.get_property("uv_index_list", [])}
+        required_udims = all_udims - selected_udims
+
+        return self._validate_udims_to_copy(task_settings, item, required_udims, selected_udims)
+
+    def _validate_udims_to_copy(self, task_settings, item, required_udims, selected_udims):
+        if not required_udims:
+            # nothing to be copied, so everything is okay
+            return True
+
+        publisher = self.parent
+
+        # find prev published version path
+        udim_copy_path_list = item.get_property("udim_copy_path_list")
+
+        if udim_copy_path_list is None:
+            item.properties["udim_copy_path_list"] = []
+
+            filters = [["entity", "is", item.context.entity],
+                       ["task", "is", item.context.task],
+                       ["name", "is", item.properties.publish_name]]
+            order = [{'field_name': 'version_number', 'direction': 'desc'}]
+
+            published_files = publisher.shotgun.find("PublishedFile",
```

Comment:
@mbrainerd Validate is now a bit slower, I'm guessing because of this statement. In comparison to the time saved by not exporting all udims, this seems alright, but need to check if we can speed this query up in any way.
Didn't want to rely of filesystem search, in case the earlier published file no longer fit the current expected template.... I know that that's unlikely to happen, but I don't want to introduce an assumption.
Let me know what you think.

Reply:
The other way is to fetch all the published files and sort by version. This is faster when we have a small number of previous versions (3-5), but the order by, limit fetch is faster when there are a larger number of versions (~22) 
The difference is about 0.01 sec either way.

Reply:
Getting the published files from Shotgun is the correct approach. I don't have a lot of input on which way is more expedient, but I do know newer versions of Shotgun (which we should be using) support significantly faster large queries. Its a paging option, though you would have to look through the python api to see what would be needed to take advantage of it.